avocado/core/nrunner/runnable.py

```diff
@@ -74,13 +77,16 @@ def __init__(self, kind, uri, *args, config=None, **kwargs):
         #: that is passed to runners, as long as a runner declares
         #: its interest in using them with
         #: attr:`avocado.core.nrunner.runner.BaseRunner.CONFIGURATION_USED`
-        self.config = config or {}
+        self._config = {}
+        self.config = config
         self.args = args
         self.tags = kwargs.pop('tags', None)
         self.dependencies = kwargs.pop('dependencies', None)
         self.variant = kwargs.pop('variant', None)
         self.output_dir = kwargs.pop('output_dir', None)
         self.kwargs = kwargs
+        self._identifier_format = config.get("runner.identifier_format",
```

Comment:
I don't quite follow what this and the adds... given that there's always logic to cover this scenario:

```python
        # For the cases where there is no config (when calling the Runnable
        # directly
        if not fmt:
            return self.uri
```

Reply:
The problem here is that the `self_config` contains only the config values based on the `STANDALONE_EXECUTABLE_CONFIG_USED` and the `runner.identifier_format` is not part of that.